src/engine/Server.cpp

```diff
@@ -841,6 +842,55 @@ Awaitable<void> Server::processQuery(
   co_return;
 }
 
+// ____________________________________________________________________________
+Awaitable<void> Server::processUpdate(
+    const ad_utility::url_parser::ParamValueMap& params, const string& update,
+    ad_utility::Timer& requestTimer,
+    const ad_utility::httpUtils::HttpRequest auto& request, auto&& send,
+    TimeLimit timeLimit) {
+  auto queryHub = queryHub_.lock();
+  AD_CORRECTNESS_CHECK(queryHub);
+  ad_utility::websocket::MessageSender messageSender{
+      getQueryId(request, update), *queryHub};
+
+  auto [cancellationHandle, cancelTimeoutOnDestruction] =
+      setupCancellationHandle(messageSender.getQueryId(), timeLimit);
+
+  auto [pinSubtrees, pinResult] = determineResultPinning(params);
+  LOG(INFO) << "Processing the following SPARQL update:"
+            << (pinResult ? " [pin result]" : "")
+            << (pinSubtrees ? " [pin subresults]" : "") << "\n"
+            << update << std::endl;
+  QueryExecutionContext qec(index_, &cache_, allocator_,
+                            sortPerformanceEstimator_, std::ref(messageSender),
+                            pinSubtrees, pinResult);
+  auto plannedQuery = co_await setupPlannedQuery(
+      params, update, qec, cancellationHandle, timeLimit, requestTimer);
+  auto qet = plannedQuery.queryExecutionTree_;
+
+  if (!plannedQuery.parsedQuery_.hasUpdateClause()) {
+    throw std::runtime_error("Expected Update but received Query.");
+  }
+
+  // TODO: activate once #1603 is merged
+  // auto& deltaTriples = index_.deltaTriplesManager();
+  auto deltaTriples = DeltaTriples(index_);
+  // TODO: will work, once #1607 is merged
+  ExecuteUpdate::executeUpdate(index_, plannedQuery.parsedQuery_, qet,
+                               deltaTriples, cancellationHandle);
+
+  LOG(INFO) << "Done processing update"
+            << ", total time was " << requestTimer.msecs().count() << " ms"
+            << std::endl;
+  LOG(DEBUG) << "Runtime Info:\n"
+             << qet.getRootOperation()->runtimeInfo().toString() << std::endl;
+  // TODO<qup42> send a proper response
+  // SPARQL 1.1 Protocol 2.2.4 Successful Responses: "The response body of a
+  // successful update request is implementation defined."
+  co_await send("");
```

Comment:
what about a 200 response with a JSON that contains
the time, the number of inserted, and probably deleted triples?
Or at least a 200 with "Update was executed successfully".

Reply:
Though that send would do that, but it seems that it doesn't. On a second thought this also wouldn't make any sense. For now I'd just add a 200 with a plain text response. Adding more detailed information and timing information is something that is on my list. Before doing that I'd like to investigate how other engines respond to update.